easygcm-lib/src/main/java/eu/inloop/easygcm/EasyGcm.java

```diff
@@ -0,0 +1,213 @@
+
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.text.TextUtils;
+
+import easygcm.R;
+
+public class EasyGcm {
+    private static final String PREFS_EASYGCM = "easygcm";
+    private static final String PROPERTY_REG_ID = "registration_id";
+    private static final String PROPERTY_APP_VERSION = "appVersion";
+
+    private static EasyGcm sInstance;
+    private GcmListener mGcmListener;
+    private GcmServicesHandler mCheckServicesHandler;
+    static volatile boolean sLoggingEnabled = true;
+
+    @SuppressWarnings("UnusedDeclaration")
+    public static void init(Context context) {
+        getInstance().onCreate(context);
+    }
+
+    synchronized static EasyGcm getInstance() {
+        if (sInstance == null) {
+            sInstance = new EasyGcm();
+        }
+        return sInstance;
+    }
+
+    private EasyGcm() {
+        mCheckServicesHandler = new GcmServicesHandler();
+    }
+
+    /**
+     * Allows to specify custom {@link eu.inloop.easygcm.GcmListener} if you don't want to implement it in the {@link
+     * android.app.Application} instance.
+     * <p/>
+     * This method should be called in {@link android.app.Application#onCreate()}.
+     *
+     * @param gcmListener custom GCM listener
+     */
+    public static void setGcmListener(GcmListener gcmListener) {
+        getInstance().mGcmListener = gcmListener;
+    }
+
+    /**
+     * Allows to specify a custom {@link eu.inloop.easygcm.GcmServicesHandler} which handles a situation
+     * when Google Play services are not available. Typically this should display a warning dialog.
+     * The default handler shows
+     * {@link com.google.android.gms.common.GoogleApiAvailability#getErrorDialog(android.app.Activity, int, int)}
+     *
+     * @param handler your custom handler for checking GcmServices.
+     */
+    public static void setCheckServicesHandler(GcmServicesHandler handler) {
+        if (handler == null) {
+            throw new IllegalArgumentException("GcmServicesHandler can't be null");
+        }
+        getInstance().mCheckServicesHandler = handler;
+    }
+
+    /**
+     * Checks, whether a registration Id for an app exists. If not, the app needs to register
+     * @param context application's context.
+     * @return true if a registration Id is persisted in shared preferences, false otherwise.
+     */
+    public static boolean isRegistered(Context context) {
+        return !TextUtils.isEmpty(getRegistrationId(context));
+    }
+
+    /**
+     * Stores the registration ID and the app versionCode in the application's
+     * {@code SharedPreferences}.
+     *
+     * @param context application's context.
+     * @param regId   registration ID
+     */
+    private static void storeRegistrationId(Context context, String regId) {
+        final int appVersion = GcmUtils.getAppVersion(context);
+        if (sLoggingEnabled) {
+            GcmUtils.Logger.d("Saving regId on app version " + appVersion);
```

Comment:
About logging - we don't like that the library logs everything even in production. We would like to set maximum logging for debug builds and only error logs for release builds. I suggest you extend library initialization with setLogLevel() method. Passing VERBOSE will log everything, passing WARNING will log only warnings and above (errors). Also this if (sLoggingEnabled) boilerplate should be removed from the code, whether to log or not should be handled on one place in one method.